test/integration/connect/test_connect.py

```diff
@@ -3088,3 +3089,94 @@ def test_set_delimiter(
     finally:
         # Stop the Instance.
         stop_app(tt_cmd, tmpdir, "test_app")
+
+
+@pytest.mark.skipif(tarantool_major_version == 1,
+                    reason="skip custom evaler test for Tarantool 1")
+def test_custom_evaler(tt_cmd, tmpdir_with_cfg):
+    tmpdir = tmpdir_with_cfg
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_output_format_app",
+                                                            "test_app.lua")
+    copy_data(tmpdir, [test_app_path])
+    start_app(tt_cmd, tmpdir, "test_app")
+
+    expected_output = '''+------------------+
+| COLUMN2          |
++------------------+
+| Hello SQL world! |
++------------------+
+'''
+    try:
+        file = wait_file(os.path.join(tmpdir, "test_app"), "ready", [])
+        assert file != ""
+
+        with open(os.path.join(tmpdir, "evaler.lua"), 'w') as f:
+            f.write("return box.execute(cmd)")
+
+        if tarantool_major_version >= 3 or (
+            tarantool_major_version == 2 and tarantool_minor_version >= 11
+        ):
+            select = "select COLUMN2 from seqscan table1 WHERE COLUMN1 = 10"
+        else:
+            select = "select COLUMN2 from table1 WHERE COLUMN1 = 10"
+
+        uris = ["localhost:3013", "tcp://localhost:3013"]
+        evalers = ["return box.execute(cmd)", os.path.join("@" + tmpdir, "evaler.lua")]
+        for uri, evaler in itertools.product(uris, evalers):
+            ret, output = try_execute_on_instance(
+                tt_cmd,
+                tmpdir,
+                uri,
+                stdin=select,
+                opts={"--evaler": evaler,
+                      "-x": "table"},
+            )
+
+            assert ret
+            assert output == expected_output
+
+    finally:
+        stop_app(tt_cmd, tmpdir, "test_app")
+
+
+def test_custom_evaler_errors(tt_cmd, tmpdir_with_cfg):
+    tmpdir = tmpdir_with_cfg
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_output_format_app",
+                                                            "test_app.lua")
+    copy_data(tmpdir, [test_app_path])
+    start_app(tt_cmd, tmpdir, "test_app")
+
+    try:
+        file = wait_file(os.path.join(tmpdir, "test_app"), "ready", [])
+        assert file != ""
+
+        with open(os.path.join(tmpdir, "evaler.lua"), 'w') as f:
+            f.write("return box.execute(cmd)")
```

Comment:
Why do you need to create a correct file for error cases? Plus it's not used here. Maybe delete this lines, or check the error for correct filepath, but incorrect command in the file.

Reply:
This is left after test code copy-paste. Removed.